mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1128,31 +1200,52 @@ def _compute_builtin_metrics(self):
         self._evaluate_sklearn_model_score_if_scorable()
         if self.model_type == _ModelType.CLASSIFIER:
             if self.is_binomial:
-                self.metrics.update(
-                    _get_binary_classifier_metrics(
-                        y_true=self.y,
-                        y_pred=self.y_pred,
-                        y_proba=self.y_probs,
-                        labels=self.label_list,
-                        pos_label=self.pos_label,
-                        sample_weights=self.sample_weights,
+                self.metrics_values.update(
+                    _get_metrics_values(
+                        _get_binary_classifier_metrics(
+                            y_true=self.y,
+                            y_pred=self.y_pred,
+                            y_proba=self.y_probs,
+                            labels=self.label_list,
+                            pos_label=self.pos_label,
+                            sample_weights=self.sample_weights,
+                        )
                     )
                 )
                 self._compute_roc_and_pr_curve()
             else:
                 average = self.evaluator_config.get("average", "weighted")
-                self.metrics.update(
-                    _get_multiclass_classifier_metrics(
-                        y_true=self.y,
-                        y_pred=self.y_pred,
-                        y_proba=self.y_probs,
-                        labels=self.label_list,
-                        average=average,
-                        sample_weights=self.sample_weights,
+                self.metrics_values.update(
+                    _get_metrics_values(
+                        _get_multiclass_classifier_metrics(
+                            y_true=self.y,
+                            y_pred=self.y_pred,
+                            y_proba=self.y_probs,
+                            labels=self.label_list,
+                            average=average,
+                            sample_weights=self.sample_weights,
+                        )
                     )
                 )
         elif self.model_type == _ModelType.REGRESSOR:
-            self.metrics.update(_get_regressor_metrics(self.y, self.y_pred, self.sample_weights))
+            self.metrics_values.update(
+                _get_metrics_values(
+                    _get_regressor_metrics(self.y, self.y_pred, self.sample_weights)
+                )
+            )
+
+    def _evaluate_builtin_metrics(self, eval_df):
+        if not self.builtin_metrics:
+            return
+        for builtin_metric in self.builtin_metrics:
+            metric_value = builtin_metric.eval_fn(eval_df, self.metrics)
+            if metric_value:
+                name = (
+                    f"{builtin_metric.name}/{builtin_metric.version}"
+                    if builtin_metric.version
+                    else builtin_metric.name
+                )
```

Comment:
/ai explain what happens if `metric_value` is 0.

Reply:
builtin metrics should always return a metric_value: MetricValue so shouldn't be an integer - I don't think this is an issue?

Reply:
What about non-builtin metrics? Do they return plaIn numbers?

Reply:
`metric_value` is None when the `eval_fn` fails to load/calculate a metric